### PR TITLE
SBERDOMA-677 Highlighting global search

### PR DESCRIPTION
**Pull request by SavelevMatthew**
global search highlighting added to contact table and employee table.

Bonuses: Can filter employee by position now + instead of empty cell displaying "—" now

### CODE REVIEW

File: apps/condo/domains/contact/hooks/useTableColumns.tsx [hunk 1]
```diff
@@ -58,6 +60,8 @@ function getFilterDropdown (message: string) {
     }
 }
 
+
+
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/contact/hooks/useTableColumns.tsx:64]:
```suggestion

```

**Review by nomerdvadcatpyat — APPROVED** [review]

File: apps/condo/domains/contact/hooks/useTableColumns.tsx [hunk 2]
```diff
@@ -58,6 +60,8 @@ function getFilterDropdown (message: string) {
     }
 }
 
+
```
Comment (pahaz) [inline apps/condo/domains/contact/hooks/useTableColumns.tsx:63]:
```suggestion
```

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/common/components/Highliter.tsx [hunk 3]
```diff
@@ -7,14 +7,18 @@ interface IHighliterProps {
     renderPart: (part: string, index: number) => React.ReactElement
 }
 
+const escapeRegex = (text: string) => {
+    return text.replace(/[-[\]{}()*+?.,\\/^$|#\s]/g, '\\$&')
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/Highliter.tsx:11]: Can we move this regexp to constants with a description of what it is doing? Maybe, someone will need it later

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/components/Highliter.tsx (hunk 3 above)
Comment (SavelevMatthew) [reply apps/condo/domains/common/components/Highliter.tsx]: done 👍

File: apps/condo/domains/contact/hooks/useTableColumns.tsx (hunk 2 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/hooks/useTableColumns.tsx]: done 👍

**Comment by SavelevMatthew** [comment]
After resolving conflict there's some errors happening....
DO NOT MERGE IT UNTIL YOU SEE THIS LINE

**Comment by SavelevMatthew** [comment]
Seems like everything ok now ;)

File: apps/condo/domains/common/components/Highliter.tsx [hunk 4]
```diff
@@ -9,12 +10,13 @@ interface IHighliterProps {
 
 export const Highliter: React.FC<IHighliterProps> = (props) => {
     const { text, search, renderPart } = props
+    if (!text) return <></>
```
Comment (pahaz) [inline apps/condo/domains/common/components/Highliter.tsx:13]: return null is better

File: apps/condo/domains/contact/hooks/useTableColumns.tsx [hunk 5]
```diff
@@ -4,6 +4,10 @@ import { useIntl } from '@core/next/intl'
 import React, { useMemo } from 'react'
 import { createSorterMap, IFilters } from '../utils/helpers'
 import { getTextFilterDropdown, getFilterIcon } from '@condo/domains/common/components/TableFilter'
+import { Highliter } from '@condo/domains/common/components/Highliter'
+import { isEmpty } from 'lodash'
+import { Typography } from 'antd'
```
Comment (pahaz) [inline apps/condo/domains/contact/hooks/useTableColumns.tsx:9]: it's better to put global import at the top

File: apps/condo/domains/common/components/Highliter.tsx (hunk 4 above)
Comment (SavelevMatthew) [reply apps/condo/domains/common/components/Highliter.tsx]: done 👍

File: apps/condo/domains/contact/hooks/useTableColumns.tsx (hunk 5 above)
Comment (SavelevMatthew) [reply apps/condo/domains/contact/hooks/useTableColumns.tsx]: done

**Pull request merged**